### PR TITLE
nuget 3.4.3 (new formula)

### PR DESCRIPTION
**Pull request by 74th**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `n` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/nuget.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  version "3.4.3"
+  sha256 "3b1ea72943968d7af6bacdb4f2f3a048a25afd14564ef1d8b1c041fdb09ebb0a"
+
+  bottle :unneeded
+
+  depends_on "mono" => :recommended
+
+  def install
+    libexec.install "nuget.exe"
+    bin.mkpath
```
Comment (bfontaine) [inline Formula/nuget.rb]: You don’t need this; `(bin/"nuget").write` will create the directory as needed.

**Comment by bfontaine** [comment]
Where does it installs packages? Will packages survive formula updates?

File: Formula/nuget.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class Nuget < Formula
+  desc "the package manager for the Microsoft development platform including .NET"
+  homepage "https://www.nuget.org/"
+  url "https://dist.nuget.org/win-x86-commandline/v3.4.3/nuget.exe"
+  version "3.4.3"
+  sha256 "3b1ea72943968d7af6bacdb4f2f3a048a25afd14564ef1d8b1c041fdb09ebb0a"
+
+  bottle :unneeded
+
+  depends_on "mono" => :recommended
```
Comment (mikemcquaid) [inline Formula/nuget.rb]: This can't run without mono so seems more than recommended?

Comment (DomT4) [inline Formula/nuget.rb]: There's some precedent for this in other Mono-using formula, IIRC. We've accepted previously people might want to use a Mono from outside of us.

File: Formula/nuget.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    libexec.install "nuget.exe"
+    bin.mkpath
+    (bin/"nuget").write <<-EOS.undent
+        #!/bin/bash
+        mono #{libexec}/nuget.exe "$@"
+    EOS
+  end
+
+  test do
+    test_str = "NuGet"
```
Comment (DomT4) [inline Formula/nuget.rb]: Might as well just inline this below given how short it is.

**Comment by 74th** [comment]
@bfontaine @mikemcquaid @DomT4 Thank for reviews.
Sorry, my commit overwrited previous commit added comments, but I fixed indicates. 
The previous commit is here https://github.com/74th/homebrew-core/commit/d6602ee1fa09211b4dc191443cd33c3997a06abe.
```diff
diff --git a/Formula/nuget.rb b/Formula/nuget.rb
index 90ddc98..cf38813 100644
--- a/Formula/nuget.rb
+++ b/Formula/nuget.rb
@@ -11,7 +11,6 @@ class Nuget < Formula

   def install
     libexec.install "nuget.exe"
-    bin.mkpath
     (bin/"nuget").write <<-EOS.undent
         #!/bin/bash
         mono #{libexec}/nuget.exe "$@"
@@ -19,7 +18,6 @@ class Nuget < Formula
   end

   test do
-    test_str = "NuGet"
-    assert_match test_str, shell_output("#{bin}/nuget help").strip
+    assert_match "NuGet", shell_output("#{bin}/nuget help").strip
   end
 end
```

File: Formula/nuget.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  sha256 "3b1ea72943968d7af6bacdb4f2f3a048a25afd14564ef1d8b1c041fdb09ebb0a"
+
+  bottle :unneeded
+
+  depends_on "mono" => :recommended
+
+  def install
+    libexec.install "nuget.exe"
+    (bin/"nuget").write <<-EOS.undent
+        #!/bin/bash
```
Comment (DomT4) [inline Formula/nuget.rb]: Style nit but we tend to start the indent in `EOS` blocks two spaces in rather than four.

File: Formula/nuget.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        mono #{libexec}/nuget.exe "$@"
+    EOS
+  end
+
+  test do
+    assert_match "NuGet", shell_output("#{bin}/nuget help").strip
```
Comment (DomT4) [inline Formula/nuget.rb]: Is the `.strip` necessary?

File: Formula/nuget.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+class Nuget < Formula
+  desc "the package manager for the Microsoft development platform including .NET"
```
Comment (DomT4) [inline Formula/nuget.rb]: Probably okay to just be:

```ruby
desc "Package manager for Microsoft development platform including .NET"
```

**Comment by DomT4** [comment]
Few style nits and would like Mike's follow-up on the `mono` dependency as it hasn't been discussed much prior, but this looks pretty solid.

**Comment by 74th** [comment]
My reason of `recommended` is other formula has it.
I installed mono framework through homebrew too, I think to remove `recommended` is a right way now.
On the other hand, Microsoft provides dnvm(.NET Version Manager), and it install mono and dotnet core, but not standard yet.

File: Formula/nuget.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  def install
+    libexec.install "nuget.exe"
+    (bin/"nuget").write <<-EOS.undent
+      #!/bin/bash
+      mono #{libexec}/nuget.exe "$@"
+    EOS
+  end
+
+  test do
+    assert_match "NuGet", shell_output("#{bin}/nuget help")
```
Comment (mikemcquaid) [inline Formula/nuget.rb]: Could you download something instead?

File: Formula/nuget.rb (hunk 2 above)
Comment (mikemcquaid) [inline Formula/nuget.rb]: Sorry for the confusion here @74th; you can change this back.

**Comment by 74th** [comment]
@mikemcquaid Yes. I added a test to search a package.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request merged**